tests/test_parser.py

```diff
@@ -1818,12 +1848,88 @@ def test_rstring():
 
     assign_stmt = statements[0]
     assert isinstance(assign_stmt, Assign)
-    assert assign_stmt.name == "a"
+    assert str(assign_stmt.name) == "a"
+    assert assign_stmt.name.location == Range("test", 2, 1, 2, 2)
     assert isinstance(assign_stmt.value, StringFormat)
 
     assign_stmt_2 = statements[1]
     assert isinstance(assign_stmt_2, Assign)
-    assert assign_stmt_2.name == "b"
+    assert str(assign_stmt_2.name) == "b"
+    assert assign_stmt_2.name.location == Range("test", 3, 1, 3, 2)
     literal = assign_stmt_2.value
     assert isinstance(literal, Literal)
     assert literal.value == r"{{a}}\n"
+
+
+def test_string_attribute_reference_1():
+    statements = parse_code(
+        r"""
+a="test{{hello.world.bye}}test"
+"""
+    )
+    assert len(statements) == 1
+    assign_stmt = statements[0]
+    assert isinstance(assign_stmt, Assign)
+    assert str(assign_stmt.name) == "a"
+    assert assign_stmt.name.location == Range("test", 2, 1, 2, 2)
+    assert isinstance(assign_stmt.value, StringFormat)
+    attribute_ref = assign_stmt.value.children[0]
+    assert attribute_ref.attribute == "bye"
+    assert attribute_ref.location == Range("test", 2, 8, 2, 26)
```

Comment:
Why is the location of bye not 3 characters long?

Reply:
Currently I create one Range for the string `{{hello.world.bye}} `and it is used both for the references as for the attributes. I could try to refine it more but I found that the code already became a lot harder to read just to create this LocatableStrings.
https://github.com/inmanta/inmanta-core/blob/583ce25bd5b332c3146b48efd6640cd5c2468e36/src/inmanta/parser/plyInmantaParser.py#L751-L761
I can try going even further if you think the current changes does not impact readability to much and if exact locations are needed.

Reply:
@sanderr

Reply:
My thoughts [here](https://github.com/inmanta/inmanta-core/pull/3778#discussion_r805711310)